### PR TITLE
Adding configuration file for OpenSUSE Leap 42.1 to lib/configfiles

### PR DESCRIPTION
**Pull request by asemen**
As the configuration files for openSUSE were dropped in past,
I would like to asks you to add this Configuration file or openSUSE Leap 42.1  named  leap_42.1.xml  to froxlor upstream project.

### CODE REVIEW

File: lib/configfiles/leap_42.1.xml [hunk 1]
```diff
@@ -0,0 +1,1474 @@
...
+						<command>
+							<visibility mode="notempty">{{settings.system.deactivateddocroot}}
+							</visibility>
+							<content><![CDATA['mkdir -p {{settings.system.deactivateddocroot}}]]></content>
+						</command>
+						<command><![CDATA[a2dismod userdir]]></command>
+					</commands>
+				</general>
+				<!-- HTTP Apache -->
+				<daemon name="apache" version="2.4" title="Apache 2.4">
```
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: missing default=true

File: lib/configfiles/leap_42.1.xml [hunk 2]
```diff
@@ -0,0 +1,1474 @@
...
+						<visibility mode="isfile">{{settings.system.apacheconf_diroptions}}
+						</visibility>
+						<content><![CDATA[echo -e 'include "{{settings.system.apacheconf_diroptions}}"' >> /etc/lighttpd/lighttpd.conf]]></content>
+					</command>
+					<command><![CDATA[lighty-disable-mod cgi]]></command>
+					<command><![CDATA[lighty-disable-mod fastcgi]]></command>
+					<command><![CDATA[service lighttpd restart]]></command>
+				</daemon>
+				<!-- HTTP Nginx -->
+				<daemon name="nginx" title="nginx">
```
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: missing acme.conf for lets encrypt (global alias) for nginx

File: lib/configfiles/leap_42.1.xml [hunk 3]
```diff
@@ -0,0 +1,1474 @@
...
+				<daemon name="bind" title="Bind9 nameserver">
+					<install><![CDATA[zypper install bind]]></install>
+					<command><![CDATA[mkdir {{settings.system.bindconf_directory}}]]></command>
+					<command><![CDATA[echo "include \"{{settings.system.bindconf_directory}}froxlor_bind.conf\";" >> /etc/named.conf]]></command>
+					<command><![CDATA[touch {{settings.system.bindconf_directory}}froxlor_bind.conf]]></command>
+					<command><![CDATA[chown named:named {{settings.system.bindconf_directory}}froxlor_bind.conf]]></command>
+					<command><![CDATA[chmod 0600 {{settings.system.bindconf_directory}}froxlor_bind.conf]]></command>
+					<command><![CDATA[systemctl restart named]]></command>
+				</daemon>
+				<daemon name="powerdns" title="PowerDNS via bind-backend">
```
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: this is now "powerdns_bind"; missing "powerdns" (standalone / gmysql)

File: lib/configfiles/leap_42.1.xml [hunk 4]
```diff
@@ -0,0 +1,1474 @@
...
+# Please check that all following paths are correct
+#
+*/5 * * * *	root	/usr/bin/nice -n 5 /usr/bin/php5 -q <BASE_PATH>scripts/froxlor_master_cronjob.php
+]]>
+						</content>
+					</file>
+					<command><![CDATA[systemctl reload cron]]></command>
+				</daemon>
+				<daemon name="libnss" title="libnss-mysql (required for FCGID/php-fpm/mpm-itk)">
+					<install><![CDATA[apt-get install libnss-mysql-bg nscd]]></install>
```
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: apt-get?

File: lib/configfiles/leap_42.1.xml [hunk 5]
```diff
@@ -0,0 +1,1474 @@
...
+						<command><![CDATA[mkdir -p {{settings.system.mod_fcgid_tmpdir}}]]></command>
+						<command><![CDATA[a2dismod php5]]></command>
+					</commands>
+					<!-- instead of just restarting apache, we let the cronjob do all the 
+						dirty work -->
+					<command><![CDATA[php {{const.FROXLOR_INSTALL_DIR}}/scripts/froxlor_master_cronjob.php --force]]></command>
+				</daemon>
+
+				<!-- PHP-FPM -->
+				<daemon name="php-fpm" title="PHP-FPM (alternative to mod_php)">
```
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: are there no commands needed for php-fpm? If no, remove this completey

**Comment by d00p** [comment]
and imho as we do not support multiple versions of opensuse, I'd suggest renaming the file to `opensuse.xml`

File: lib/configfiles/leap_42.1.xml (hunk 1 above)
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: missing acme.conf for lets encrypt (global alias) for apache

File: lib/configfiles/leap_42.1.xml [hunk 6]
```diff
@@ -0,0 +1,1474 @@
...
+        Require env REDIRECT_STATUS
+    </Location>
+</IfModule>
+]]>
+						</content>
+					</file>
+					<command><![CDATA[service apache2 restart]]></command>
+				</daemon>
+				<!-- HTTP Lighttpd -->
+				<daemon name="lighttpd" title="LigHTTPd">
```
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: missing acme.conf for lets encrypt (global alias) for lighttpd

File: lib/configfiles/leap_42.1.xml (hunk 1 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: ok were to be added?

File: lib/configfiles/leap_42.1.xml (hunk 2 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: ok in which line?

File: lib/configfiles/leap_42.1.xml (hunk 3 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: ok I will change to powerdns_bind gmysql I have to look up

File: lib/configfiles/leap_42.1.xml (hunk 4 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: sure will be changed

File: lib/configfiles/leap_42.1.xml (hunk 5 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: will be removed

File: lib/configfiles/leap_42.1.xml (hunk 3 above)
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: see for example https://github.com/Froxlor/Froxlor/blob/master/lib/configfiles/gentoo.xml#L391 and following lines (that's gentoo, needs adjusting)

File: lib/configfiles/leap_42.1.xml (hunk 1 above)
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: see https://github.com/Froxlor/Froxlor/blob/master/lib/configfiles/gentoo.xml#L99 (change pathes accordingly)

File: lib/configfiles/leap_42.1.xml (hunk 6 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: I agree to change it to opensuse.xml

File: lib/configfiles/leap_42.1.xml (hunk 2 above)
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: See https://github.com/Froxlor/Froxlor/blob/master/lib/configfiles/gentoo.xml#L270 (change pathes accordingly)

File: lib/configfiles/leap_42.1.xml (hunk 6 above)
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: see https://github.com/Froxlor/Froxlor/blob/master/lib/configfiles/gentoo.xml#L176

File: lib/configfiles/leap_42.1.xml (hunk 1 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: sorry I do not understadn waht to do?
Please help.
Were should be default=true  added?

File: lib/configfiles/leap_42.1.xml (hunk 2 above)
Comment (asemen) [inline lib/configfiles/leap_42.1.xml]: I have the line                                         <file name="/etc/nginx/acme.conf">
in section 
"                                "
should this line be somewhere else.
I do not understand what is wrong

File: lib/configfiles/leap_42.1.xml (hunk 1 above)
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: `<daemon name="apache" version="2.4" title="Apache 2.4" default="true">`

File: lib/configfiles/leap_42.1.xml (hunk 2 above)
Comment (d00p) [inline lib/configfiles/leap_42.1.xml]: You are missing the complete `<file>` block for the acme.conf, right before `<file name="/etc/nginx/fastcgi_params">` (Line 214) add: 

`					<file name="/etc/nginx/acme.conf">
						<visibility mode="true">{{settings.system.leenabled}}
						</visibility>
						<content><![CDATA[
location /.well-known/acme-challenge {
	alias {{settings.system.letsencryptchallengepath}}/.well-known/acme-challenge;
	location ~ /.well-known/acme-challenge/(.*) {
		default_type text/plain;
	}
}
]]>
						</content>
					</file>`

**Comment by asemen** [comment]
Did the wanted change adds  related to comments

**Comment by d00p** [comment]
Leap is now 2 years old, recommend either doing this for current 15.1 (2019-10-03) or close this as I am not very familiar with openSuse and cannot maintain that

**Comment by asemen** [comment]
Lets close this. It tooks to long.

**Pull request closed without merging**